### DISCUSSION TITLE
How to combine custom mapping and automatic mapping generation from query ?

### DISCUSSION THREAD

**Discussion by k94ll13nn3** [alirezanet/Gridify#123] [Q&A] [answered]
Hello,

First thanks for this project, it is really useful.

I would like to be able to use the automatic generation of mapping (from the `FixMapper` method I saw in the repo) when using the `Gridify` method with custom maps added for some properties.

Using the sample from the documentation, I would like to be able to do this withtout specifiyng `Contact.PhoneNumber`. If I don't use a custom mapper, the mapping on `Contact.PhoneNumber` works, but if I use a custom mapper without `Contact.PhoneNumber`, it does not work anymore (in the sample I could use `Contact.Address` but my use cases are not as straighforward).
``` csharp
var mapper = new GridifyMapper<Person>()
            .GenerateMappings()
            .AddMap("address", p => p.Contact.Address)
            .AddMap("Contact.PhoneNumber", p => p.Contact.PhoneNumber); // I would like to remove this
```

Is it possible for now without having to respecify all possible mapping for basic parent.child properties ?

**Comment by alirezanet** [comment]
Hi @k94ll13nn3,
Currently, if you define a custom mapper you have to generate the mappings manually for the child classes, 
I'll convert this discussion to an issue and look into it further, it is a good improvement point. 

you can follow the progress related to this feature in issue #124

**Comment by alirezanet** [comment, marked as answer]
Hi @k94ll13nn3,
This feature has been added to `v2.11.0`.  _(EDIT: please use `v2.11.1` instead)_
Please let me know if you face any problems.

More info:
[https://alirezanet.github.io/Gridify/guide/gridifyMapper.html#generatemappings](https://alirezanet.github.io/Gridify/guide/gridifyMapper.html#generatemappings)

**Reply by k94ll13nn3** [reply]
Thanks a lot, I will look at it 👍

**Reply by k94ll13nn3** [reply]
I just tested it, and it seems that the generated mapping for nested properties are generated for the top level class, and it throws
```
using Gridify;

Console.WriteLine(new GridifyMapper<Person>().GenerateMappings(0)); // userName,firstName,lastName,password
Console.WriteLine(new GridifyMapper<Person>().GenerateMappings(1)); // throws Instance property 'Address' is not defined for type 'Person'

public class Person
{
    public string UserName { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Password { get; set; }
    public Contact Contact { get; set; }

}

public class Contact
{
    public string Address { get; set; }
    public int PhoneNumber { get; set; }
}

```

**Reply by alirezanet** [reply]
Hi @k94ll13nn3,
Thank you for bringing this up, I missed one edge case in the previous version, it should be fixed in `v2.11.1`

**Reply by k94ll13nn3** [reply]
All seems to be good, thank you!
